### PR TITLE
Updated cake to version 3.1.0 and corrected breaking changes cake code

### LINKED ISSUE #4465: Update cake version in build scripts
There's a few draft PRs in progress right now that have had to upgrade cake in the build scripts as part of other changes. It might be nice to upgrade cake separately as part of this issue to make those and other future updates simpler

### PR DESCRIPTION
**Pull request by OsirisTerje**
While correcting the script after the update, I noticed that we do have some tasks that seems to not be used,  Publish Test results is one - that one seems to be oriented to Azure Pipelines.  And then SignPackages.  This one I believed we should have there, but I guess we might need to update it, and also ensure we have a valid certificate.
On the other hand, nuget does an automatic repository signing for the packages. So, do we need more?

Fixes #4465

### CODE REVIEW

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on:
     branches:
       - master
       - release
-      - 'v3.13-dev'
+      - 'v3'
```
Comment (manfred-brands) [inline .github/workflows/continuous_integration.yml:8]: Our `release` branch is pointing to `v3.13.2`. Should that be moved or deleted?

File: build.cake [hunk 2]
```diff
@@ -109,15 +107,15 @@ Task("Build")
...
     .Does(() =>
     {
-        DotNetCoreBuild(SOLUTION_FILE, CreateDotNetCoreBuildSettings());
+        DotNetBuild(SOLUTION_FILE, CreateDotNetCoreBuildSettings());
     });
 
-DotNetCoreBuildSettings CreateDotNetCoreBuildSettings() =>
-    new DotNetCoreBuildSettings
+DotNetBuildSettings CreateDotNetCoreBuildSettings() =>
+    new DotNetBuildSettings
```
Comment (manfred-brands) [inline build.cake:114]: For consistency, drop `Core` from the method name as well.

File: build.cake [hunk 3]
```diff
@@ -109,15 +107,15 @@ Task("Build")
...
     {
         Configuration = configuration,
         NoRestore = true,
-        Verbosity = DotNetCoreVerbosity.Minimal
+        Verbosity = DotNetVerbosity.Minimal
```
Comment (manfred-brands) [inline build.cake:118]: I had added the below to ensure the version is part of the dll.
```cake
MSBuildSettings = new DotNetMSBuildSettings
{
    Version = packageVersion
}
```

This was previously done on AppVeyor builds only, but that was removed.

File: src/NUnitFramework/tests/Constraints/CollectionEquivalentConstraintTests.cs [hunk 4]
```diff
@@ -287,8 +287,10 @@ public void WorksWithNonIComparableTuples()
 
         // The following tests are each running in 14ms to 46ms on my machine. Based on that,
         // warn at 100ms and fail at 500ms
-        private const int LARGE_COLLECTION_WARN_TIME = 100;
-        private const int LARGE_COLLECTION_FAIL_TIME = 500;
+        // Seems to be slower on MacOs on build on github actions, so increasing this with 50%
+
+        private const int LARGE_COLLECTION_WARN_TIME = 100 * 3 / 2;
+        private const int LARGE_COLLECTION_FAIL_TIME = 500 * 3 / 2;
```
Comment (manfred-brands) [inline src/NUnitFramework/tests/Constraints/CollectionEquivalentConstraintTests.cs:293]: If only MaCOS is the problem, don't run them there:
`[Platform(Exclude = "MACOSX", Reason = "MacOS github runner is very slow")]`

**Review by manfred-brands — CHANGES_REQUESTED** [review]
Some suggestions.

File: .github/workflows/continuous_integration.yml (hunk 1 above)
Comment (OsirisTerje) [reply .github/workflows/continuous_integration.yml:8]: The release branch is not in direct parent path behind v3, so I would prefer to just skip it.

Comment (OsirisTerje) [reply .github/workflows/continuous_integration.yml:8]: Deleted the rule now, and also deleted the branch itself

File: build.cake (hunk 2 above)
Comment (OsirisTerje) [reply build.cake:114]: Done

File: build.cake (hunk 3 above)
Comment (OsirisTerje) [reply build.cake:118]: Done

File: src/NUnitFramework/tests/Constraints/CollectionEquivalentConstraintTests.cs (hunk 4 above)
Comment (OsirisTerje) [reply src/NUnitFramework/tests/Constraints/CollectionEquivalentConstraintTests.cs:293]: Done

File: build.cake (hunk 3 above)
Comment (OsirisTerje) [reply build.cake]: But it has a property of that type, so moved there

**Review by manfred-brands — APPROVED** [review]
Thanks. Nothing further from me.

**Review by stevenaw — APPROVED** [review]
Thanks @OsirisTerje , especially for the added macos fix. Looks good.

**Pull request merged**